doc/neps/nep-0044-restructuring-numpy-docs.rst

```diff
@@ -0,0 +1,234 @@
+- Geospatial data (e.g. lat/lon/time to create maps for every year via a stacked
+  array, like `gridMet data <http://www.climatologylab.org/gridmet.html>`_)
+- Using text data and dtypes (e.g. use speeches from different people, shape
+  ``(n_speech, n_sentences, n_words)``)
+
+The *Preparing to Teach* document [2]_ from the Software Carpentry Instructor
+Training materials is a nice summary of how to write effective lesson plans (and
+tutorials would be very similar). In addition to adding new tutorials, we also
+propose a *How to write a tutorial* document, which would help users contribute
+new high-quality content to the documentation.
+
+Data sets
+---------
+
+Using interesting data in the NumPy docs requires giving all users access to
+that data, either inside NumPy or in a separate package. The former is not the
+best idea, since it's hard to do without increasing the size of NumPy
+significantly. Even for SciPy there has so far been no consensus on this (see
+`scipy PR 8707 <https://github.com/scipy/scipy/pull/8707>`_ on adding a new
+``scipy.datasets`` subpackage).
+
+So we'll aim for a new (pure Python) package, named ``numpy-datasets`` or
+``scipy-datasets`` or something similar. That package can take some lessons from
+how, e.g., scikit-learn ships data sets. Small data sets can be included in the
+repo, large data sets can be accessed via a downloader class or function.
+
+Related Work
+============
+
+Some examples of documentation organization in other projects:
+
+- `Documentation for Jupyter <https://jupyter.org/documentation>`_
+- `Documentation for Python <https://docs.python.org/3/>`_
+- `Documentation for TensorFlow <https://www.tensorflow.org/learn>`_
+
+These projects make the intended audience for each part of the documentation
+more explicit, as well as previewing some of the content in each section. 
+
+Implementation
+==============
+
+Besides rewriting the current documentation to some extent, it would be ideal
+to have a technical infrastructure that would allow more contributions from the
+community. For example, if Jupyter Notebooks could be submitted as-is as
+tutorials or How-Tos, this might create more contributors and broaden the NumPy
+community.
+
+Similarly, if people could download some of the documentation in Notebook
+format, this would certainly mean people would use less outdated material for
+learning NumPy.
+
+It would also be interesting if the new structure for the documentation makes
+translations easier.
+
+Currently, the `documentation for NumPy <https://numpy.org/devdocs/>`_ can be
+confusing, especially for beginners. Our proposal is to reorganize the docs in
+the following structure:
+
+- For users:
+    - Absolute Beginners Tutorial
+    - main Tutorials section
+    - How To's for common tasks with NumPy
+    - Reference Guide
+    - Explanations
+    - F2Py Guide
+    - Glossary
+- For developers/contributors:
```

Comment:
Do API docs or under-the-hood docs fit in here somewhere?

Reply:
API docs are the same as "Reference Guide". @melissawm maybe change to `Reference Guide (API reference)` for clarity?

under-the-hood docs is a good suggestion, there are definitely topics like that that we already have docs for or that would be useful. If it helps users understand things, then I'd say it falls under Explanations. If it's developer-focused (e.g. how to add a new ufunc), then we may need a new category because "Contributor Guide" is meant to be more about the process of contributing (setting up a dev environment, style guide, etc.). I don't have a great name for that category to suggest; maybe add "Under-the-hood docs" as a placeholder for now?

Reply:
I added under-the-hood docs for now.